asmcomp/arm64/emit.mlp

```diff
@@ -954,7 +969,10 @@ let emit_instr env i =
     | Lop(Ireturn_addr) ->
         let n = frame_size env in
         if env.f.fun_frame_required then
-          `	ldr	{emit_reg i.res.(0)}, [sp, #{emit_int (n-8)}]\n`
+          if fp then
+            `	ldr	{emit_reg i.res.(0)}, [sp, #{emit_int (n-16)}]\n`
+          else
+            `	ldr	{emit_reg i.res.(0)}, [sp, #{emit_int (n-8)}]\n`
```

Comment:
That chunk is incorrect - `Ireturn_addr` loads the return address saved on stack. Or arm64, this is `x30`, and it is always saved at `[sp + frame size - 8]`.